Example/src/screens/HeaderOptions.tsx

```diff
@@ -0,0 +1,181 @@
+  Settings: undefined;
+};
+
+interface MainScreenProps {
+  navigation: NativeStackNavigationProp<StackParamList, 'Main'>;
+}
+
+const MainScreen = ({navigation}: MainScreenProps): JSX.Element => {
+  return (
+    <ScrollView>
+      <Button
+        onPress={() => navigation.navigate('Settings')}
+        title="Go to next screen"
+      />
+      <Button onPress={() => navigation.pop()} title="🔙 Back to Examples" />
+    </ScrollView>
+  );
+};
+
+interface SettingsScreenProps {
+  navigation: NativeStackNavigationProp<StackParamList, 'Settings'>;
+}
+
+const SettingsScreen = ({navigation}: SettingsScreenProps): JSX.Element => {
+  const [headerTitle, setHeaderTitle] = useState('Settings');
+  const [backButtonVisible, setBackButtonVisible] = useState(true);
+  const [headerShown, setHeaderShown] = useState(true);
+  const [headerLargeTitle, setHeaderLargeTitle] = useState(false);
+  const [headerItem, setHeaderItem] = useState('right');
+  const [headerBackTitle, setHeaderBackTitle] = useState('Back');
+  const [headerHideShadow, setHeaderHideShadow] = useState(false);
+  const [headerTranslucent, setHeaderTranslucent] = useState(false);
+
+  const square = (props: {tintColor?: string}) => (
+    <Square {...props} color="green" size={20} />
+  );
+
+  useLayoutEffect(() => {
+    navigation.setOptions({
+      headerTitle,
+      headerBackTitleVisible: backButtonVisible, // iOS
+      headerHideBackButton: !backButtonVisible, // android
+      headerShown,
+      headerRight: headerItem === 'right' ? square : undefined,
+      headerCenter: headerItem === 'center' ? square : undefined,
+      headerLeft: headerItem === 'left' ? square : undefined,
+      headerLargeTitle, // iOS
+      headerBackTitle, // iOS
+      headerHideShadow,
+      headerTranslucent,
+    });
+  }, [
+    navigation,
+    headerTitle,
+    backButtonVisible,
+    headerItem,
+    headerShown,
+    headerLargeTitle,
+    headerBackTitle,
+    headerHideShadow,
+    headerTranslucent,
+  ]);
+
+  return (
+    <SafeAreaView style={styles.container}>
+      <SettingsInput
+        label="Header title"
+        value={headerTitle}
+        onValueChange={setHeaderTitle}
+      />
+      <SettingsSwitch
+        label="Back button visible"
+        value={backButtonVisible}
+        onValueChange={setBackButtonVisible}
+      />
+      <SettingsSwitch
+        label="Header shown"
+        value={headerShown}
+        onValueChange={setHeaderShown}
+      />
+      <SettingsSwitch
+        label="Header hide shadow"
+        value={headerHideShadow}
+        onValueChange={setHeaderHideShadow}
+      />
+      <SettingsSwitch
+        label="Header translucent"
+        value={headerTranslucent}
+        onValueChange={setHeaderTranslucent}
+      />
+      <SettingsPicker
+        label="Header item"
+        value={headerItem}
+        onValueChange={setHeaderItem}
+        items={['left', 'center', 'right']}
+      />
+      {Platform.OS === 'ios' ? (
```

Comment:
I think we can make it visible on Android, but with the note of `iOS only`, so changing it will not be effective.